src/cryptography/x509/extensions.py

```diff
@@ -48,8 +41,7 @@ def _key_identifier_from_public_key(public_key):
             serialization.PublicFormat.SubjectPublicKeyInfo
         )
 
-        spki = _SubjectPublicKeyInfo.load(serialized)
-        data = spki["subjectPublicKey"].cast(OctetBitString).native
+        data = six.binary_type(PublicKeyInfo.load(serialized)['public_key'])
```

Comment:
What I was suggesting here is that `PublicKeyInfo` has an attribute `.sha1` which returns the SHA1 of the public key bytes. It would effectively replace this line and line 46 – there would be no reason to call `hashlib.sha1(data).digest()`.